### DISCUSSION TITLE
[v5] Custom exception

### DISCUSSION THREAD

**Discussion by steschwa** [adonisjs/core#2061] [Help] [answered]
I'm using below @adonis package versions:

```
@adonisjs/assembler: "^3.0.7",
@adonisjs/auth: "^5.0.2",
@adonisjs/core: "^5.0.4-preview-rc-2",
@adonisjs/lucid: "^9.0.3",
@adonisjs/session: "^4.0.5"
```

I want to create a custom exception which i'll throw if a query parameter is missing.
So `node ace make:exception MissingQuery` will create a class `MissingQueryException`. Unfortunately this class extends from `Exception` imported from `@adonis/core` which Typescript complains about

> **File ..../node_modules/@adonisjs/core/build/adonis-typings/index.d.ts is not a module**

In the AdonisJs Docs the `Exception` class is imported from `@poppinss/utils` (https://preview.adonisjs.com/guides/http/exception-handling#self-handled-exceptions).

Should i install `@poppinss/util` or are the typings missing in `@adonis/core`?

**Comment by iagobruno** [comment, marked as answer]
I had the same problem and I solved it by changing the import of @adonisjs/core module by @poppinss/utils:
```diff
- import { Expeption } from '@adonisjs/core'
+ import { Exception } from '@poppinss/utils'
...
```

Poppinss is already installed with Adonis.

**Comment by thetutlage** [comment]
Recently had a discussion with @targos and we will expose the `Exception` class from the `@adonisjs/core` in the next release. But for now, you can rely on the `@poppinss/utils` module.
